src/Client.php

```diff
@@ -1186,6 +1186,39 @@ public function imageSearch($term)
         )->then(array($this->parser, 'expectJson'));
     }
 
+    /**
+     * Create a new image from a container
+     *
+     * @param string $container The ID or name of the container to commit
+     * @param string $repo      Repository name for the created image
+     * @param string $tag       Tag name for the created image
+     * @param string $comment   Commit message
+     * @param string $author    Author of the image (e.g., John Hannibal Smith <hannibal@a-team.com>)
+     * @param bool   $pause     Whether to pause the container before committing, default: true
+     * @param string $changes   Dockerfile's instructions to apply while committing
+     * @param array  $config    The container configuration (body of the request)
+     * @return PromiseInterface
+     * @link https://docs.docker.com/engine/api/v1.41/#operation/ImageCommit
+     */
+    public function containerCommit($container, $repo = null, $tag = null, $comment = null, $author = null, $pause = true, $changes = null, $config = array())
```

Comment:
* Docblocks should mention `null` values like `@param ?string $repo` etc. (see also other params)
* Last parameter type should be `array $config = array()` (other types would require PHP 7.1+)

Reply:
Added nullable to the docblock. But I don't quite understand the second point should'nt the type be omitted for backwards compatibility?

Reply:
Full type support would indeed require PHP 7.1+, but the `array` type declaration is already supported in older PHP versions: https://3v4l.org/12MCi

Reply:
Oh okay I didn't know that thanks I will add it to the PR 👍🏼

Reply:
Updated